### ISSUE TITLE
Remove `cache` field from TreeWalker in Checkstyle 8.0

### ISSUE THREAD

**Issue by MEZk** [checkstyle/checkstyle#2883]
base on problems at https://github.com/checkstyle/checkstyle/pull/2882#issuecomment-181031148

The field 'cache' and 'setCacheFile' from TreeWalker now are used by maven-checkstyle-plugin.
We need to remove 'cache' field and method 'setCacheFile' from TreeWalker in Checkstyle 8.0 since they were moved to Checker in accordance with https://github.com/checkstyle/checkstyle/issues/569.

maven plugin already moved to use cache on Checker - https://github.com/apache/maven-plugins/blob/trunk/maven-checkstyle-plugin/src/main/java/org/apache/maven/plugins/checkstyle/exec/DefaultCheckstyleExecutor.java#L388
fix was done at 2016-10-20 - https://github.com/apache/maven-plugins/commit/17c9ffe1e2a8f1c52ac220f2fd3ca291cb6af7d1
last release is 2.17 from 2015-10-15 - https://maven.apache.org/plugins/maven-checkstyle-plugin/history.html

method is already empty, so we have to just wait for new release of checkstyle maven plugin

**Comment by rnveach** [comment]
@romani This looks like a Checkstyle 8 candidate.

**Comment by subkrish** [comment]
I'm on it.

**Comment by romani** [comment]
@subkrish , our CI is failed due to removed property - it is not acceptable.
what we should todo:
1) create PRs to all project that we use in our CI to move cache to Checker - share PR to them here.
2) If "1)" is taking time to merge on affected projects ..... we need to keep setter method that is deprecated and do nothing at all (just to not fail with exception)
3) ones we release and 1) is merged we will remove TreeWalker empty setter for property completely.

**Comment by rnveach** [comment]
@romani Comment at https://github.com/checkstyle/checkstyle/pull/4471#issuecomment-310836769 says we can't release until maven is updated, but wercker failure at https://github.com/checkstyle/checkstyle/pull/4471#issuecomment-309586591 says it was problem with `Postgresql JDBC Driver` configuration.

Why do we need to wait on maven when we can just make changes to the configuration to pass?
We don't support cache in TreeWalker anyways, so it doesn't make a difference if it is removed now.

**Comment by romani** [comment]
we depend on maven plugin only due to the fact we use it regression testing(sh, groovy, .....).
If non of CI fails ... we need to recheck that we actually substitute version to use. If we do substitution and all CI pass - we good to do any breaking changes now.

**Comment by rnveach** [comment]
I was wrong. It is still a maven-plugin issue. We can't continue until they upgrade to 2.18.
Their issue is at https://issues.apache.org/jira/browse/MCHECKSTYLE-332 .

**Comment by rnveach** [comment]
We have a new issue for asking `maven-checkstyle-plugin` to make a release at https://issues.apache.org/jira/browse/MCHECKSTYLE-346 .
This issue can't be completed until either they release a new version or we completely break away from using their plugin.

See https://github.com/checkstyle/contribution/issues/273 , https://github.com/checkstyle/checkstyle/issues/5385 , https://github.com/checkstyle/checkstyle/issues/5386 .

**Comment by romani** [comment]
fix is merged.
